### ISSUE TITLE
Bug(no-duplicate-schema-methods): support array-of-arrays

### ISSUE THREAD

**Issue by derekcicerone** [marcalexiei/eslint-plugin-zod#344]
### 📦 Affected package

eslint-plugin-zod

### 🧩 Description

The latest release is now flagging code like this:
```
z.string().nonempty().array().array()
```

We are working around by switching to this:
```
z.array(z.array(z.string().nonempty()))
```

### 🪜 How to reproduce

See above.

### ✅ Expected behavior

Probably shouldn't flag this pattern for this rule (though I will acknowledge the new code we had to use as a workaround is more clear).

### 💻 System info

```shell
N/A
```

**Comment by marcalexiei** [comment]
Hello Derek,

Thanks for opening the issue!

> Probably shouldn't flag this pattern for this rule

Yes, you're right. This pattern should not be reported by this rule.

A PR to fix the issue is already in place, and I plan to merge it shortly.

---

> (though I will acknowledge the new code we had to use as a workaround is more clear)

If you prefer the function syntax, you may want to enable the array-style rule. By default, it reports usages of chained array methods.

[See the array-style documentation for more information.](https://github.com/marcalexiei/eslint-zod/blob/main/plugins/eslint-plugin-zod/docs/rules/array-style.md)

**Comment by derekcicerone** [comment]
Neat, thanks for the pointer - I’ll check that rule out.

**Comment by marcalexiei** [comment]
Fix available in [eslint-plugin-zod@4.7.0](https://github.com/marcalexiei/eslint-zod/releases/tag/eslint-plugin-zod%404.7.0).

Thanks again for the report!
